### DISCUSSION TITLE
Trailing whitespace removed in raw text

### DISCUSSION THREAD

**Discussion by adiabatic** [typst/typst#3111] [Q&A] [answered]
I have a bit of text:

```typst
#show raw: (it) => {
  show: highlight.with(fill: luma(80%))
  it
}

- padded: ` one two three `
```

However, the gray background doesn't extend for one space past the `three` in the raw block like it does in the source:

![Screenshot 2023-12-31 at 3 35 40 PM](https://github.com/typst/typst/assets/101246/93b88735-31f9-4859-96ac-57153f975191)

Is there a way to fix this, or is this a bug?

**Comment by 8LWXpg** [comment, marked as answer]
That's because trailing whitespaces is trimmed by `raw`. you can add zero width space at the end of the string.

```typst
#show raw: (it) => {
  show: highlight.with(fill: luma(80%))
  it + sym.zws
}

- padded: ` one two three `
```

![1](https://github.com/typst/typst/assets/105704427/a77d38dd-4bbc-4090-ad28-b96a216556c1)

**Comment by laurmaedje** [comment]
It's a bug: #597
